git-secrets-scan-action/README.md

```diff
@@ -0,0 +1,42 @@
+# GitHub Action to `git secrets --scan` in your CI  🔄
+
+This simple action uses the [vanilla AWS CLI](https://docs.aws.amazon.com/cli/index.html) and [git secrets](https://github.com/awslabs/git-secrets) to scan your repository for secrets being commited to public. 
+
+The action checks for common AWS patterns and ensures that keys present in ~/.aws/credentials are not found in any commit. The following checks are added:
+
+* AWS Access Key IDs via (A3T[A-Z0-9]|AKIA|AGPA|AIDA|AROA|AIPA|ANPA|ANVA|ASIA)[A-Z0-9]{16}
+* AWS Secret Access Key assignments via ":" or "=" surrounded by optional quotes
+* AWS account ID assignments via ":" or "=" surrounded by optional quotes
+* Allowed patterns for example AWS keys (AKIAIOSFODNN7EXAMPLE and wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY)
+* Known credentials from ~/.aws/credentials
+
+## Usage
+
+### `workflow.yml` Example
+
+Place in a `.yml` file such as this one in your `.github/workflows` folder. [Refer to the documentation on workflow YAML syntax here.](https://help.github.com/en/articles/workflow-syntax-for-github-actions)
+
+```
+name: Scan Secrets
+on: push
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v1
+    - name: scan-git-secrets
+      uses: aws-robotics/aws-robomaker-github-actions/git-secrets-scan-action@2.4.0
+```
+
+
+### Environment Variables
+
+| Key | Value | Type | Required |
+| ------------- | ------------- | ------------- | ------------- |
+| `GITHUB_WORKSPACE` | The GitHub workspace directory path. The workspace directory contains a subdirectory with a copy of your repository if your workflow uses the actions/checkout action. If you don't use the actions/checkout action, the directory will be empty. For example, /home/runner/work/my-repo-name/my-repo-name. This variable is set by GH actions. | `env` | **Yes** |
```

Comment:
I think we should check if the directory pointed to by this environment variable is correct. Currently, if this environment variable is incorrect, the action will pass by default, right? But we want it to fail in that case.

Reply:
A correct directory should be at least a non-empty path string that points to an existent path.

Reply:
I believe GH actions guarantees that this env variable is set.. but a good catch to point that the default behavior here (when the path is empty/incorrect) would be that the action passes with exit code 0.

Reply:
Added a check to see if GITHUB_WORKSPACE is set and the repo is checked-out.. Tested that the action now fails in both cases..

Reply:
Also, `GITHUB_WORKSPACE` can be set explicitly for this step in the workflow, right? I ask because sometimes `actions/checkout` may be used multiple times (so does `GITHUB_WORKSPACE` end up taking on the value of the path that we want) and/or not used to check out the sources that we want to actually scan (for example, the way `ros-tooling/action-ros-ci` is used).

Reply:
`GITHUB_WORKSPACE` can be explicitly set although the user might not know the path to the workspace inside the docker container which GH actions set by default. Do you also mean that some other action would update this env variable and then it would not point to the correct directory?

Per https://github.com/actions/checkout#Checkout-multiple-repos-side-by-side, any checkout action occurs within `GITHUB_WORKSPACE` by default. (or `GITHUB_WORKSPACE/path` if a path is provided in actions/checkout). 
 
reference from actions/checkout: 
```
# Relative path under $GITHUB_WORKSPACE to place the repository
    path: ''
```
But now it looks like we are not accommodating for that `path` variable.

Reply:
Added `path` input variable to accommodate the repository being checked out at a different location within `GITHUB_WORKSPACE`

Reply:
While testing out the action I figured that, 
In actions/checkout@v1, `path` is the absolute path where the repo will get checked-out.
In actions/checkout@v2, `path` is relative to `GITHUB_WORKSPACE` where the repo will get checked-out.

So now I guess the best way for us to deal with this is use absolute path to the repo. A issue with that being the user might not know the path to the workspace inside the docker container which GH actions set by default( via `GITHUB_WORKSPACE`)

Reply:
While testing out the action I figured that, 
In actions/checkout@v1, `path` is the absolute path where the repo will get checked-out.
In actions/checkout@v2, `path` is relative to `GITHUB_WORKSPACE` where the repo will get checked-out.

A way out of this is to check for both, local path and absolute path. If neither point to a valid directory, we exit with failure.

Reply:
While testing out the action I figured that, 
In actions/checkout@v1, input variable `path` is not used. 
In actions/checkout@v2, input variable `path` is relative to `GITHUB_WORKSPACE` where the repo will get checked-out.

A potential way out of this is to check for both, local path and absolute path. If neither point to a valid directory, we exit with failure. 

We can always ask the user for the absolute path to the repo, however, if the repo is checked-out wrt `GITHUB_WORKSPACE`, the user might not know the absolute path.

Reply:
While testing out the action I figured that, 
In actions/checkout@v1, input variable `path` is not used. 
In actions/checkout@v2, input variable `path` is relative to `GITHUB_WORKSPACE` where the repo will get checked-out.

A potential way out of this is to check for the path per `actions/checkout@v1` or per `actions/checkout@v2` based on if `'path` is set or not. If the path does not point to a valid directory, exit with failure. This however, would *probably* not work in the case of `actions/checkout@v2` being used for nested repositories.

As a simple alternative, we can always ask the user for the absolute path to the repo, however, if the repo is checked-out wrt `GITHUB_WORKSPACE`, the user might not know the absolute path.

Reply:
I made a relevant comment at https://github.com/aws-robotics/aws-robomaker-github-actions/pull/17#discussion_r504952382.